meshroom/ui/qml/Viewer3D/DefaultCameraController.qml

```diff
@@ -81,17 +81,39 @@ Entity {
                 return
             }
             if (zooming) { // zoom with alt + RMD
-                d = (root.camera.viewCenter.minus(root.camera.position)).length() * 0.1
-                var tz = axisMX.value * root.translateSpeed * d
                 mouseHandler.hasMoved = true
-                root.camera.translate(Qt.vector3d(0, 0, tz).times(dt), Camera.DontTranslateViewCenter)
+                d = root.camera.viewCenter.minus(root.camera.position).length() // Distance between camera position and center position
+                var tz = axisMX.value * root.translateSpeed * 0.2 // Translation to apply depending on user action (mouse move), bigger absolute value means we'll zoom/dezoom more
+
+                // We forbid too big zoom, as it means the distance between camera and center would be too low and we'll have no translation after (due to float representation)
+                if (tz >= 0.9 * d)
+                    return
+
+                // We forbid too small zoom as it means we are getting very close to center position and next zoom may lead to similar problem as previous cases (no translation)
+                if (tz > 0 && tz <= 0.001)
+                    return
+
+                root.camera.translate(Qt.vector3d(0, 0, tz), Camera.DontTranslateViewCenter)
                 return
             }
         }
+
         onDoubleClicked: mouseDoubleClicked(mouse)
         onWheel: {
-            var d = (root.camera.viewCenter.minus(root.camera.position)).length() * 0.2
-            var tz = (wheel.angleDelta.y / 120) * d
+            var d = root.camera.viewCenter.minus(root.camera.position).length() // Distance between camera position and center position
+            var tz = (wheel.angleDelta.y / 120) * d * 0.2 // Translation to apply depending on user action (mouse wheel), bigger absolute value means we'll zoom/dezoom more
+
+            // We forbid too big zoom, as it means the distance between camera and center would be too low and we'll have no translation after (due to float representation)
+            if (tz >= 0.9 * d) {
+                console.warn("zoom too much")
+                return
+            }
+
+            // We forbid too small zoom as it means we are getting very close to center position and next zoom may lead to similar problem as previous cases (no translation)
+            if (tz > 0 && tz <= 0.001) {
```

Comment:
Looks to me we should test the distance d directly !!  ("it means we are getting very close to center position") why using some function of d when d is ok ?

Reply:
I tested with d, but the bug still occurs (the condition tz > 0 cannot be replaced with d > 0), so I kept tz